sk_cathode/utils/evaluation_functions.py

```diff
@@ -365,3 +508,63 @@ def preds_2d(data, data_preds, labels=None, bins=30, ranges=None,
         plt.savefig(out_file)
     if not suppress_show:
         plt.show()
+
+
+def profile_1d(data, preds, bins):
+    """
+    Helper function to compute the average predictions for the data
+    in each bin, in 1D.
+    """
+    hist, edges = np.histogram(data, bins=bins)
+    avg_preds = np.zeros_like(hist, dtype=float)
+    avg_preds[:] = np.nan
+
+    for i in range(len(hist)):
+        mask = (data >= edges[i]) & (data < edges[i+1])
+        if np.any(mask):
+            avg_preds[i] = np.mean(preds[mask])
+
+    return avg_preds, edges
+
+
+def profile_2d(data_x, data_y, preds, bins):
+    """
+    Helper function to compute the average predictions for the data
+    in each bin, in 2D.
+    """
+
+    hist, xedges, yedges = np.histogram2d(data_x, data_y, bins=bins)
+    avg_preds = np.zeros_like(hist)
+    avg_preds[:] = np.nan
+
+    for i in range(hist.shape[0]):
+        for j in range(hist.shape[1]):
+            mask = (data_x >= xedges[i]) & (data_x < xedges[i+1]) & \
+                   (data_y >= yedges[j]) & (data_y < yedges[j+1])
+            if np.any(mask):
+                avg_preds[i, j] = np.mean(preds[mask])
+
+    return avg_preds, xedges, yedges
+
+
+def rescale(x, range=(0.45, 0.55)):
+    """
+    Helper function to rescale the input to the range [0, 1].
```

Comment:
This is slightly confusing, at first I thought the `range` parameter was the desired output range, but it is the input range.

Reply:
Good point, it is renamed it `input_range` now.